python/pylibcudf/pylibcudf/strings/combine.pyx

```diff
@@ -0,0 +1,223 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+from libcpp.memory cimport unique_ptr
+from libcpp.utility cimport move
+from pylibcudf.column cimport Column
+from pylibcudf.libcudf.column.column cimport column
+from pylibcudf.libcudf.scalar.scalar cimport string_scalar
+from pylibcudf.libcudf.scalar.scalar_factories cimport (
+    make_string_scalar as cpp_make_string_scalar,
+)
+from pylibcudf.libcudf.strings cimport combine as cpp_combine
+from pylibcudf.scalar cimport Scalar
+from pylibcudf.table cimport Table
+
+from cython.operator import dereference
+from pylibcudf.libcudf.strings.combine import \
+    output_if_empty_list as OutputIfEmptyList  # no-cython-lint
+from pylibcudf.libcudf.strings.combine import \
+    separator_on_nulls as SeparatorOnNulls  # no-cython-lint
+
+
+cpdef Column concatenate(
+    Table strings_columns,
+    ColumnOrScalar separator,
+    Scalar narep=None,
+    Scalar col_narep=None,
+    separator_on_nulls separate_nulls=separator_on_nulls.YES,
+):
+    """
+    Concatenate all strings in the column into one new string delimited
+    by an optional separator string.
```

Comment:
```suggestion
    Concatenate all columns in the table horizontally into one new string delimited
    by an optional separator string.
```
?

Since there's a table as input I'm not sure what "strings in the column" means.